### PR TITLE
Rails4 Compatability

### PR DESCRIPTION
**Pull request by cantino**
@shirish-pampoorickal @randycoulman @urnf 

It's not clear to be why we would need `unscoped` / `with_exclusive_scope` on `parent`.

### CODE REVIEW

**Comment by shirish-pampoorickal** [comment]
LGTM!

**Comment by cantino** [comment]
@shirish-pampoorickal Do you have any idea why `with_exclusive_scope` was used?

**Comment by bgreg** [comment]
@shirish-pampoorickal @cantino @urnf 

We had tests on Rails 4 failing due to the fact that `populate_materialized_path` was trying to grab path_string off parents that didn't exist even though parent_id was set on the child.  This checks that both the parent and parent_id exist first.

Please ensure that this expectation is correct since we are changing it slightly.  We've added a test for it.

**Comment by cantino** [comment]
@shirish-pampoorickal Opinion on this?

**Comment by shirish-pampoorickal** [comment]
@cantino I did not find any reason for using `with_exclusive_scope`. @urnf @bgreg LGTM

**Comment by tmccrmck** [comment]
@shirish-pampoorickal @cantino 

Anyone opposed to merging this?

File: lib/arboreal/instance_methods.rb [hunk 1]
```diff
@@ -83,9 +83,7 @@ def sibling_conditions
 
     def populate_materialized_path
       if parent_id_changed? || materialized_path.nil?
-        model_base_class.send(:with_exclusive_scope) do
-          self.materialized_path = parent ? parent.path_string : "-"
-        end
+        self.materialized_path = (parent_id && parent) ? parent.path_string : "-"
```
Comment (cantino) [inline lib/arboreal/instance_methods.rb]: Is it possible for `parent` to exist but not have `parent_id`, such as during object building with new records?

**Comment by cantino** [comment]
I'm out next week -- @shirish-pampoorickal owns this.

**Comment by shirish-pampoorickal** [comment]
@cantino @naiyt @trentmccormick This LGTM!

**Comment by cantino** [comment]
Go ahead and merge, and maybe only bump Gemfile in rails 4?

**Comment by cantino** [comment]
@naiyt Let's merge but not bump on master, only on rails 4.

**Comment by naiyt** [comment]
Sounds good to me.

**Comment by cantino** [comment]
Ok, since the branch is deleted, be sure to bundle update on rails4.

**Pull request merged**